java/runtime/src/main/java/io/ray/runtime/metric/Gauge.java

```diff
@@ -0,0 +1,73 @@
+package io.ray.runtime.metric;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+
+public class Gauge implements Metric {
+
+  private String name;
+  private double value;
+  private long gaugePtr = 0L;
+  Map<TagKey, String> tags;
+
+  public Gauge(String name, String description, String unit, Map<TagKey, String> tags) {
+    gaugePtr = registerGaugeNative(name, description, unit,
+      tags.keySet().stream().mapToLong(TagKey::getNativePointer).toArray());
+    this.name = name;
+    this.tags = tags;
+  }
+
+  private native long registerGaugeNative(String name, String description,
+                                          String unit, long[] nativeTagKeyPtrList);
+  private native void unregisterGauge(long gaugePtr);
```

Comment:
Why not `unregisterGaugeNative` just like `registerGaugeNative `?

Reply:
> Why not `unregisterGaugeNative` just like `registerGaugeNative `?

JNI will return native pointer when registering gauge, so unregisterGauge need this native pointer.